src/frontend/src/handler/create_source.rs

```diff
@@ -112,6 +132,20 @@ pub async fn handle_create_source(
                 pk_column_ids: pk_column_ids.into_iter().map(Into::into).collect(),
             }
         }
+        SourceSchema::Avro(avro_schema) => {
+            assert_eq!(columns.len(), 1);
```

Comment:
Is this assertion meaningful or only for temporary purposes?

Reply:
it aligns with create source with pb row format. I think we can remove it after pk is fully supported.